src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -0,0 +1,172 @@
+            elif not self.get_argument_enable_open_telemetry_traces() and DATA_DOG_DEST in existing_traces:
+                existing_traces.remove(DATA_DOG_DEST)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "tracesConfiguration", "destinations", value=existing_traces)
+
+    def set_up_open_telemetry_metrics_destinations(self, existing_managed_env_def):
+        existing_metrics = safe_get(existing_managed_env_def, "properties", "openTelemetryConfiguration", "metricsConfiguration", "destinations")
+        if existing_metrics is None:
+            if self.get_argument_enable_open_telemetry_metrics():
+                existing_metrics = [DATA_DOG_DEST]
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "metricsConfiguration", "destinations", value=existing_metrics)
+        else:
+            if self.get_argument_enable_open_telemetry_traces() and DATA_DOG_DEST not in existing_metrics:
+                existing_metrics.append(DATA_DOG_DEST)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "metricsConfiguration", "destinations", value=existing_metrics)
+            elif not self.get_argument_enable_open_telemetry_traces() and DATA_DOG_DEST in existing_metrics:
+                existing_metrics.remove(DATA_DOG_DEST)
+                safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "metricsConfiguration", "destinations", value=existing_metrics)
+
+    def update(self):
+        try:
+            return self.client.update(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                                      name=self.get_argument_name(), managed_environment_envelope=self.managed_env_def, no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+class ContainerappEnvTelemetryAppInsightsPreviewSetDecorator(BaseResource):
+
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.managed_env_def = {}
+    
+    def get_argument_open_telemetry_app_insights_connection_string(self):
+        return self.get_param("connection_string")
+    
+    def get_argument_enable_open_telemetry_traces(self):
+        return self.get_param("enable_open_telemetry_traces")
+    
+    def get_argument_enable_open_telemetry_logs(self):
+        return self.get_param("enable_open_telemetry_logs")
+
+    def construct_payload(self):
+        try:
+            r = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except CLIError as e:
+            handle_raw_exception(e)
+
+        # General setup
+        safe_set(self.managed_env_def, "location", value=r["location"])  # required for API
+
+        existing_managed_env_def = None
```

Comment:
Recommend to set existing_managed_env_def  as a property of Decorator, using like: `self.existing_managed_env_def `

`self.existing_managed_env_def = None`